docs/MachineConfiguration.md

```diff
@@ -180,7 +180,8 @@ When a machine boots with `nosmt` Kernel Argument, it disables multi-threading o
 ### KernelType
 
 This feature is available with OCP 4.4 and onward releases as both `day 1` and `day 2` operation. It allows to choose between traditional and Real Time (RT) kernel on an RHCOS node. Supported values are
-`""` or `default` for traditional kernel and `realtime` for RT kernel.
+`""` or `default` for traditional kernel, `realtime` for RT kernel and `64k-pages` for 64k memory pages on aarch64.
+Note that `64k-pages` and `realtime` cannot be selected at the same time. Also, 64k pages support is limited to aarch64 architecture.
```

Comment:
is there some way we can limit this change to apply only on aarch64 nodes? and possibly throw some sort of error if the associated machine pool is not aarch64 based?

Reply:
As I know, RT kernel is not supported on ARM64 cluster, So apply kernelType `realtime` + `64k-pages` on ARM64 cluster is invalid case.

If we apply kernelType `64k-pages` on AMD64 cluster, machine config pool will be degraded with below error message 
```console
- lastTransitionTime: '2023-10-18T06:55:53Z'
  message: 'Failed to render configuration for pool worker: kernelType=64k-pages is
    invalid'
  reason: ''
  status: 'True'
  type: RenderDegraded
```

Reply:
@Prashanth684  the MCD will throw an error. See https://github.com/openshift/machine-config-operator/pull/3903/files#diff-349c0748c3f52201852d3027c29daf618b45300b949482728df6666a3f9ba245R1145-R1149

Reply:
oops..did not notice that ...thanks

Reply:
also does this error mean that the machine will be unusable when the update fails? i am thinking of a case where we have a multi-arch compute cluster with x86+arm64 compute nodes? would the x86 machines error out and be in a "NotReady" state?

Reply:
@Prashanth684 I'll try it out today and followup

Reply:
> Note that `64k-pages` and `realtime` cannot be selected at the same time. Also, 64k pages support is limited to aarch64 architecture.

Should we also say that `realtime` cannot be selected on `aarch64` to make things clearer (not sure for P/Z)?

Reply:
> also does this error mean that the machine will be unusable when the update fails? i am thinking of a case where we have a multi-arch compute cluster with x86+arm64 compute nodes? would the x86 machines error out and be in a "NotReady" state?

If the `realtime` kernel is applying on arm64 node, the machine will be degraded i.e. `machineconfiguration.openshift.io/state: Degraded` annotation `machineconfiguration.openshift.io/reason` will show you that rt packages are not available

Reply:
Clusterbot is not working for me today so I can't try that quickly without going through the process of building a whole release payload, which I won't be able to do today.

Reply:
If user applies unsupported combination node will goes degraded. MCO applies config per pool. So, it is admin responsibility to make sure that they apply generic config when a pool has multi architecture nodes. Good thing is if one node fails to apply the config, MCD marks the node degraded and hence it won't start upgrading another node (when maxUnavialble: 1) to cascade the issue. On the degraded node, existing stuff would work fine but in order to schedule a new pod, issue on the node will need to be first resolved so that update can complete and node is marked again schedulable.